react/features/room-lock/middleware.js

```diff
@@ -0,0 +1,36 @@
+/* global APP */
+import JitsiMeetJS from '../base/lib-jitsi-meet';
+
+import { CONFERENCE_FAILED } from '../base/conference';
+import { MiddlewareRegistry } from '../base/redux';
+import { _showPasswordDialog } from './actions';
+
+/**
+ * Middleware that captures conference failed and checks for password required
+ * error and requests a dialog for user to enter password.
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+
+    switch (action.type) {
+    case CONFERENCE_FAILED: {
```

Comment:
I don't think this is possible, seeing password dialog means you are not in a muc, so technically nothing is going on.

This PR, at least the web part, will not be merged as it will lead to inconsistency in UI.